src/main/java/org/opensearch/security/dlic/rest/api/SecuritySSLCertsAction.java

```diff
@@ -0,0 +1,338 @@
+                                  final RestController controller,
+                                  final Client client,
+                                  final AdminDNs adminDNs,
+                                  final ConfigurationRepository cl,
+                                  final ClusterService cs,
+                                  final PrincipalExtractor principalExtractor,
+                                  final PrivilegesEvaluator privilegesEvaluator,
+                                  final ThreadPool threadPool,
+                                  final AuditLog auditLog,
+                                  final SecurityKeyStore securityKeyStore,
+                                  final boolean certificatesReloadEnabled) {
+        super(settings, configPath, controller, client, adminDNs, cl, cs, principalExtractor, privilegesEvaluator, threadPool, auditLog);
+        this.securityKeyStore = securityKeyStore;
+        this.restApiAdminPrivilegesEvaluator =
+                new RestApiAdminPrivilegesEvaluator(threadPool.getThreadContext(), privilegesEvaluator, adminDNs);
+        this.certificatesReloadEnabled = certificatesReloadEnabled;
+        this.httpsEnabled = settings.getAsBoolean(SSLConfigConstants.SECURITY_SSL_HTTP_ENABLED, true);
+    }
+
+    @Override
+    public List<Route> routes() {
+        return ROUTES;
+    }
+
+    @Override
+    protected void handleApiRequest(final RestChannel channel, final RestRequest request, final Client client) throws IOException {
+        switch (request.method()) {
+            case GET:
+                if (!restApiAdminPrivilegesEvaluator.isCurrentUserRestApiAdminFor(getEndpoint(), "certs")) {
+                    forbidden(channel, "");
+                    return;
+                }
+                handleGet(channel, request, client, null);
+                break;
+            case PUT:
+                if (!restApiAdminPrivilegesEvaluator.isCurrentUserRestApiAdminFor(getEndpoint(), "reloadcerts")) {
+                    forbidden(channel, "");
+                    return;
+                }
+                if (!certificatesReloadEnabled) {
+                    badRequestResponse(
+                            channel,
+                            String.format(
+                                    "no handler found for uri [%s] and method [%s]. In order to use SSL reload functionality set %s to true",
+                                    request.path(),
+                                    request.method(),
+                                    ConfigConstants.SECURITY_SSL_CERT_RELOAD_ENABLED
+                            )
+                    );
+                    return;
+                }
+                handlePut(channel, request, client, null);
+                break;
+            default:
+                notImplemented(channel, request.method());
+                break;
+        }
+    }
+
+    /**
+     * GET request to fetch transport certificate details
```

Comment:
nit: Can the json be formatted in this comment similar to how it was in `SecuritySSLCertsInfoAction`?